src/main/java/app/SearchUseCaseFactory.java

```diff
@@ -1,46 +1,47 @@
-//import interface_adapter.search.SearchController;
-//import interface_adapter.search.SearchViewModel;
-//import use_case.search.SearchInputBoundary;
-//import use_case.search.SearchInteractor;
-//import use_case.search.SearchOutputBoundary;
-//import use_case.search.SearchUserDataAccessInterface;
-//import view.SearchView;
-//
-//import javax.swing.*;
-//import javax.swing.text.View;
-//import java.io.IOException;
-//
-//public class SearchUseCaseFactory {
-//    /** Prevent instantiation. */
-//    private SearchUseCaseFactory() {}
-//
-//    public static SearchView create(
-//            ViewManagerModel viewManagerModel, SearchViewModel searchViewModel, SearchUserDataAccessInterface searchUserDataAccessObject) {
-//
-//        try {
-//            SearchController searchController = createSearchUseCase(viewManagerModel, searchViewModel, searchUserDataAccessObject);
-//            return new SearchView(searchController, searchViewModel);
-//        } catch (IOException e) {
-//            JOptionPane.showMessageDialog(null, "Could not open user data file.");
-//        }
-//
-//        return null;
-//    }
-//
-//    private static SearchController createSearchUseCase(ViewManagerModel viewManagerModel, SearchViewModel signupViewModel, SearchUserDataAccessInterface userDataAccessObject) throws IOException {
-//
-//        // Notice how we pass this method's parameters to the Presenter.
-//        SearchOutputBoundary searchOutputBoundary = new SearchPresenter(viewManagerModel, signupViewModel);
-//
-//        SearchInputBoundary searchSignupInteractor = new SearchInteractor(
-//                userDataAccessObject, searchOutputBoundary);
-//
-//        return new SearchController(searchSignupInteractor);
-//    }
-//
-//}
+// Author: André
+
+package app;
+
+import entity.UniversityFactory;
+import interface_adapter.ViewManagerModel;
+import interface_adapter.search.SearchController;
+import interface_adapter.search.SearchPresenter;
+import interface_adapter.search.SearchViewModel;
+import use_case.search.SearchInputBoundary;
+import use_case.search.SearchInteractor;
+import use_case.search.SearchOutputBoundary;
+import use_case.search.SearchUserDataAccessInterface;
+import view.SearchView;
+
+import javax.swing.*;
+import java.io.IOException;
+
+public class SearchUseCaseFactory {
+    /** Prevent instantiation. */
+    private SearchUseCaseFactory() {}
+
+    public static SearchView create(
+            ViewManagerModel viewManagerModel, SearchViewModel searchViewModel, SearchUserDataAccessInterface searchUserDataAccessObject, UniversityFactory shortUniversityFactory) {
+
+        try {
+            SearchController searchController = createSearchUseCase(viewManagerModel, searchViewModel, searchUserDataAccessObject, shortUniversityFactory);
+            return new SearchView(searchController, searchViewModel);
+        } catch (IOException e) {
+            JOptionPane.showMessageDialog(null, "Could not open user data file.");
```

Comment:
Eventually, maybe switch this to some sort of a facade/strategy, so that we can have many different messages that all go through the same show message dialogue. Idea would be we pass just a message variable instead of a hard coded message. Also, this message seems to no longer be relevant here, since we aren't opening user data files, instead, maybe "error accessing API" or something more specific